python/pyspark/sql/dataframe.py

```diff
@@ -4430,6 +4430,48 @@ def withColumnRenamed(self, existing: str, new: str) -> "DataFrame":
         """
         return DataFrame(self._jdf.withColumnRenamed(existing, new), self.sparkSession)
 
+    def withColumnsRenamed(self, *colsMap: Dict[str, str]) -> "DataFrame":
```

Comment:
We also need to add a new API to `python/docs/source/reference/pyspark.sql/dataframe.rst` for documentation.

Reply:
Do we generate this file or this has to be curated by hand?

Reply:
I have added entry to rst file. Not sure if that is enough.

Reply:
Cool. It's enough!

You can run the command `make clean html` to generate the documents in `python/docs` path, if you want to make sure the documentation display properly :-)

Reply:
> Do we generate this file or this has to be curated by hand?

Adding the API to the existing `rst` file should be enough. (That's exactly what you did 👍 )

Reply:
@itholic thank you for the review!